polygeist/tools/cgeist/Lib/clang-mlir.cc

```diff
@@ -69,41 +72,80 @@ static cl::opt<bool>
 
 constexpr llvm::StringLiteral MLIRASTConsumer::DeviceModuleName;
 
-bool isLLVMStructABI(const RecordDecl *RD, llvm::StructType *ST) {
-  if (!CombinedStructABI)
-    return true;
-  if (RD->isUnion())
+/******************************************************************************/
+/*                             CodeGenUtils                                   */
+/******************************************************************************/
+
+mlir::IntegerAttr CodeGenUtils::wrapIntegerMemorySpace(unsigned memorySpace,
+                                                       MLIRContext *ctx) {
+  return memorySpace ? mlir::IntegerAttr::get(mlir::IntegerType::get(ctx, 64),
+                                              memorySpace)
+                     : nullptr;
+}
+
+bool CodeGenUtils::isAggregateTypeForABI(QualType qt) {
```

Comment:
is this function new / used?

Reply:
IS new and is not used in this PR but will be used in PR #6994.

Reply:
Can you please change the title and/or description of this PR to mention that? other than that LGTM.